x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/EsPhysicalOperationProviders.java

```diff
@@ -173,7 +173,7 @@ private static class DefaultShardContextForUnmappedField extends DefaultShardCon
     }
 
     public Function<org.elasticsearch.compute.lucene.ShardContext, Query> querySupplier(QueryBuilder builder) {
-        QueryBuilder qb = builder == null ? QueryBuilders.matchAllQuery() : builder;
+        QueryBuilder qb = builder == null ? QueryBuilders.matchAllQuery().boost(0.0f) : builder;
```

Comment:
I wonder if there is a follow up optimisation we could do here.
In the case where no `QueryBuilder` is available and we are using `match_all`, when we are adding the score block:
- we would no longer need to retrieve the scores from Lucene (we know they should all be zero), so we no longer need to set the score mode to `COMPLETE`
- we could be using `ConstantDoubleVector` to set the score block, which I _guess_ would be using a constant amount of memory, no matter how many docs we retrieve.

Reply:
> we would no longer need to retrieve the scores from Lucene (we know they should all be zero), so we no longer need to set the score mode to COMPLETE

I believe a similar optimization is done by the MatchAllDocsQuery internally - it uses a `ConstantScoreScorer` which I don't think uses the COMPLETE score mode.

> we could be using ConstantDoubleVector to set the score block, which I guess would be using a constant amount of memory, no matter how many docs we retrieve.

Yes, that's doable. Right now the Lucene operators do not take that into account. We can do it as a follow up.